engine/access/rpc/backend/connection_factory.go

```diff
@@ -138,46 +139,70 @@ func (cf *ConnectionFactoryImpl) retrieveConnection(grpcAddress string, timeout
 	return conn, nil
 }
 
-func (cf *ConnectionFactoryImpl) GetAccessAPIClient(address string) (access.AccessAPIClient, error) {
+func (cf *ConnectionFactoryImpl) GetAccessAPIClient(address string) (access.AccessAPIClient, io.Closer, error) {
 
 	grpcAddress, err := getGRPCAddress(address, cf.CollectionGRPCPort)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	conn, err := cf.retrieveConnection(grpcAddress, cf.CollectionNodeGRPCTimeout)
+	var conn *grpc.ClientConn
+	if cf.ConnectionsCache != nil {
+		conn, err = cf.retrieveConnection(grpcAddress, cf.CollectionNodeGRPCTimeout)
+		if err != nil {
+			return nil, nil, err
+		}
+		return access.NewAccessAPIClient(conn), nil, err
```

Comment:
here you could return a noop closer:
```golang
type noopCloser struct{}
func (c *noopCloser) Close() {}
```

```suggestion
		return access.NewAccessAPIClient(conn), &noopCloser{}, err
```

that reduces that amount of boilerplate logic you need in all the callers since we always return a real closer implementation